std/algorithm/sorting.d

```diff
@@ -212,6 +250,58 @@ bool isSorted(alias less = "a < b", Range)(Range r) if (isForwardRange!(Range))
     assert(isSorted(s));   // bidirectional
 }
 
+// isStrictlySorted
+/**
+Checks whether a forward range is strictly sorted according to the comparison
+operation $(D less). Performs $(BIGOH r.length) evaluations of $(D
+less).
+
+Params:
+    less = Predicate the range should be strictly sorted by.
+    r = Forward range to check for sortedness.
+
+Returns: true if the range is strictly sorted, false otherwise.
+*/
+bool isStrictlySorted(alias less = "a < b", Range)(Range r) if (isForwardRange!(Range))
+{
+    return isSortedImpl!(less, true)(r);
+}
+
+///
+@safe unittest
+{
+    assert([1, 2].isStrictlySorted);
+    assert(![1, 1, 2].isStrictlySorted);
+    assert([2, 1].isStrictlySorted!"a > b");
```

Comment:
Honestly exactly for such short comparison I like string lambda.
However changed to actual lambdas.

Maybe you should add that to the style guide ;-)